### DISCUSSION TITLE
Videos not playing on iPhone and download not working

### DISCUSSION THREAD

**Discussion by Notesuser** [immich-app/immich#11251] [Q&A]
Hello everyone,

I have been using immich for several months now and everything is going very well. I use an Android mobile phone and pictures and videos load quickly at home and on the move.

Now I have a visitor and wanted to use immich to share the pictures.  Uploading and sharing albums works perfectly. But my visitor has an iPhone and he can't see or download videos either on his home wifi or while travelling. When trying to watch videos, sometimes a black or grey screen appears, sometimes the first frame of the video and if it's going well, you can hear the sound, but the picture is still. All videos are converted to VP9 on the server. With video downloads, the circle goes round and round.

I don't think this behaviour is due to immich. But unfortunately I don't know anything about the iPhone. Since I use the immich distribution and not the container, my server is still on version v.1.08.0, the apps on Android and iOS are version v1.109.0.

I am grateful for any advice. Many thanks in advance

**Comment by alextran1502** [comment]
VP9 format doesn't get supported by the video player on iPhone. i think that is the reason

**Reply by Notesuser** [reply]
Thank you very much for your answer.

That's a shame, because the help for selecting the codec says ‘VP9 has high efficiency and web compatibility, but takes longer to transcode.’ I read up on this and would never have thought that the iphone couldn't handle it - and ultimately only because VP9 comes from Google.

I will now switch to hevc and test whether it works.

Thanks again for your help.

**Reply by lukehmcc** [reply]
Did you end up getting a solution to this? I'm in a very similar situation, but I have everything to be encoded to h264... so that really doesn't make sense if that isn't playing on an iphone.

**Reply by Viperuss2009** [reply]
> Did you end up getting a solution to this? I'm in a very similar situation, but I have everything to be encoded to h264... so that really doesn't make sense if that isn't playing on an iphone.

Use hevc for iOS devices. In my experience only hevc working with them

**Reply by dotfortun3-code** [reply]
I am having the same issue, but encoding is set to hevc. I can't watch in the app or the browser, all I get is audio. This only affects videos in a shared library. If it's my videos, it works. If it's my wife's it is black with audio only.

Edit: I should add, all of the videos are shot using iPhones and are already hevc, so they aren't getting transcoded which makes it really weird that they do not play anything but audio.

**Comment by lukehmcc** [comment]
Okay so what seems to have fixed it for me. I disabled hardware encoding and re-encoded everything to h264. Very annoying as if they download the videos they can play them, but hardware encoded videos won't play for them in the immich app. Oh well.
